### ISSUE TITLE
Add building mapping tool for square and rectangular buildings

### ISSUE THREAD

**Issue by aawiseman** [hotosm/tasking-manager#5974]
**Is your feature request related to a problem? Please describe.**
Right now mappers need to manually square buildings, and often don't, leading to a lot of effort and discussion to fix the issue. I think iD in the TM should have a separate building tool, in addition to point, line and polygon. 

**Describe the solution you'd like**
A new tool on the toolbar at the top for building - it's automatically rectilinear and tagged building=yes. Similar to the existing JOSM building tool. 

**Describe alternatives you've considered**
Nothing is really effective -- warnings, after the fact checks, comments, etc.

**Additional context**

**Comment by aawiseman** [comment]
Right now a major issue people have with mapping from the HOT TM is unsquared buildings, and HOT has a major effort to identify and fix those. This would let you map a square building (or rectangle) in the first place, resulting in better initial data quality

**Comment by SColchester** [comment]
Releated to this long-running iD issue: https://github.com/openstreetmap/iD/issues/528

**Comment by HelNershingThapa** [comment]
Related PR - https://github.com/openstreetmap/iD/pull/2699

**Comment by aawiseman** [comment]
Looks like that PR was not done? or something, unclear to me

**Comment by ramyaragupathy** [comment]
Yeah - it was not completed. Let me follow up w/ Quincy and Bryan to have additional context.

**Comment by RAytoun** [comment]
I and many other validators would really like to see this tool added to the iD Editor to reduce the number of unsquared buildings that keep on happening.
